test/basicfuns.jl

```diff
@@ -44,6 +44,34 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for z in (0, 0.0, 0f0, false, -Inf)
+        @test iszero(@inferred xexpx(z))
+    end
+    for nan in (NaN16, NaN32, NaN64)
+        @test isnan(@inferred xexpx(nan))
```

Comment:
Could also check that the type is as expected:
```suggestion
        @test @inferred(xexpx(nan)) === nan
```